xbrowse_server/base/model_utils.py

```diff
@@ -103,42 +134,50 @@ def find_matching_seqr_model(xbrowse_model):
                 individual_id=xbrowse_model.indiv_id)
         elif xbrowse_class_name == "ProjectTag":
             return xbrowse_model.seqr_variant_tag_type if xbrowse_model.seqr_variant_tag_type else SeqrVariantTagType.objects.get(
-                project__deprecated_project_id=xbrowse_model.project.project_id,
-                name=xbrowse_model.title)
+                Q(project__deprecated_project_id=xbrowse_model.project.project_id) | Q(project__isnull=True),
+                Q(name=xbrowse_model.tag))
         elif xbrowse_class_name == "VariantTag":
             if xbrowse_model.seqr_variant_tag:
                 return xbrowse_model.seqr_variant_tag
 
+            criteria = {
+                'variant_tag_type__name': xbrowse_model.project_tag.tag,
+                'saved_variant__project__deprecated_project_id': xbrowse_model.project_tag.project.project_id,
+                'saved_variant__xpos_start': xbrowse_model.xpos,
+                'saved_variant__ref': xbrowse_model.ref,
+                'saved_variant__alt': xbrowse_model.alt,
+            }
             if xbrowse_model.family:
-                return SeqrVariantTag.objects.get(
-                    variant_tag_type__project__deprecated_project_id=xbrowse_model.project_tag.project.project_id,
-                    xpos_start=xbrowse_model.xpos,
-                    ref=xbrowse_model.ref,
-                    alt=xbrowse_model.alt,
-                    family__family_id=xbrowse_model.family.family_id)
-            else:
-                return SeqrVariantTag.objects.get(
-                    variant_tag_type__project__deprecated_project_id=xbrowse_model.project_tag.project.project_id,
-                    xpos_start=xbrowse_model.xpos,
-                    ref=xbrowse_model.ref,
-                    alt=xbrowse_model.alt)
+                criteria['saved_variant__family__family_id'] = xbrowse_model.family.family_id
+            return SeqrVariantTag.objects.get(**criteria)
+        elif xbrowse_class_name == "VariantFunctionalData":
+            if xbrowse_model.seqr_variant_functional_data:
+                return xbrowse_model.seqr_variant_functional_data
+
+            criteria = {
+                'functional_data_tag': xbrowse_model.functional_data_tag,
+                'saved_variant__xpos_start': xbrowse_model.xpos,
+                'saved_variant__ref': xbrowse_model.ref,
+                'saved_variant__alt': xbrowse_model.alt,
+            }
+            if xbrowse_model.family:
+                criteria['saved_variant__family__family_id'] = xbrowse_model.family.family_id
+                criteria['saved_variant__project__deprecated_project_id'] = xbrowse_model.family.project.project_id
```

Comment:
because for VariantTag the project is coming from the project_tag which is always there. For VariantFunctionalData there isn't such a constraint, the only place project is stored is on family, which is optional so we can only add it if family is defined